src/firstboot/firstboot.c

```diff
@@ -308,28 +308,36 @@ static int prompt_locale(void) {
         return 0;
 }
 
-static int process_locale(void) {
-        const char *etc_localeconf;
+static int process_locale(int rfd) {
+        _cleanup_close_ int pfd = -EBADF;
         char* locales[3];
         unsigned i = 0;
         int r;
 
-        etc_localeconf = prefix_roota(arg_root, "/etc/locale.conf");
-        if (laccess(etc_localeconf, F_OK) >= 0 && !arg_force) {
-                log_debug("Found %s, assuming locale information has been configured.",
-                          etc_localeconf);
+        r = chase_symlinks_at(rfd, "etc", CHASE_AT_RESOLVE_IN_ROOT|CHASE_NONEXISTENT, NULL, &pfd);
```

Comment:
CHASE_NONEXISTENT sounds wrong

Reply:
I don't think it is, /etc/ could be a symlink to a nonexistent directory. In that case, we should probably create it at that location, which means we need CHASE_NONEXISTENT to resolve the symlink.

I did change the logic to save the resolved path and to always create it with `open_mkdir_at_p()` instead of `open_mkdir_at()`.